src/components/entry/EntryNav.js

```diff
@@ -1,7 +1,70 @@
-const EntryNav = () => (
-  <>
-    <p>EntryNav</p>
-  </>
-)
+import { ArrowBack } from '@styled-icons/boxicons-regular'
+import { Pencil } from '@styled-icons/boxicons-solid'
+import { MapMarkerAlt as Map } from '@styled-icons/fa-solid'
+import styled from 'styled-components'
+
+import { theme } from '../ui/GlobalStyle'
+import IconButton from '../ui/IconButton'
+
+const EntryNavContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+`
+
+const EntryNavTextContainer = styled.div`
+  display: flex;
+  align-items: center;
+  flex-direction: row;
+`
+
+const EntryNavText = styled.p`
+  margin-left: 10px;
+`
+
+const EntryNavIconsContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  width: 110px;
+`
+
+const EntryNav = () => {
+  const onEditEntryDetailsButtonClick = () => {
+    // TODO: edit entry callback
+    console.log('Edit entry details clicked')
+  }
+
+  const onViewEntryOnMapButtonClick = () => {
+    // TODO: view entry on map callback
+    console.log('View entry on map clicked')
+  }
+
+  return (
+    <EntryNavContainer>
+      <EntryNavTextContainer>
+        <ArrowBack size="20" color={theme.secondaryText} />
```

Comment:
Also tip on implementing the back button. Since we have routing you should be able to just `history.pop`. That'll take you back to the right view, regardless if it was map or list.